### ISSUE TITLE
[no-unnecessary-condition] Map#get(key: K) is creating a false positive

### ISSUE THREAD

**Issue by aaomidi** [typescript-eslint/typescript-eslint#1635]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "warn"
  }
}
```

```TS
const map: Map<string, Object> = new Map<string, Object>();


function testRule(key: string): void {
	const value = map.get(key);
	if (value) {
		console.log('value exists');
	}
}
```

**Expected Result**

This should not trigger the rule

**Actual Result**

```
  6:6  warning  Unnecessary conditional, value is always truthy  @typescript-eslint/no-unnecessary-condition

✖ 1 problem (0 errors, 1 warning)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.21.0` |
| `@typescript-eslint/parser`        | `2.21.0` |
| `TypeScript`                       | `3.8.2` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Are you using `strictNullChecks`?

**Comment by aaomidi** [comment]
@bradleyayers No

**Comment by bradzacher** [comment]
Without `strictNullChecks`, `null` and `undefined` are essentially erased from the types.
Which means the lint rule sees the type as just `Object`, not `Object | undefined`.

You must use the compiler option.
